pandas/core/dtypes/dtypes.py

```diff
@@ -547,6 +548,8 @@ def update_dtype(self, dtype: "CategoricalDtype") -> "CategoricalDtype":
             ).format(dtype=dtype)
             raise ValueError(msg)
 
+        dtype = cast(CategoricalDtype, dtype)
```

Comment:
Can you move this below subsequent comment? Seems to read better and more logically fit there

Reply:
As an aside I was kind of curious why this couldn't be inferred more easily since it's really just a str and a CategoricalDtype that are allowed here. So ideally I thought `isinstance(str)` and `isinstance(self)` would be two possible branches.

I *think* the reason this wasn't laid out that way is to allow `Categorical` objects to be passed as well, but doing so actually causes an error:

```python
t = pd.CategoricalDtype(categories=['b', 'a'], ordered=True)
c = pd.Categorical(['a','b','c','a','b','c'], ordered=True, categories=['c', 'b', 'a'])
>>> t.update_dtype(c)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/williamayd/clones/pandas/pandas/core/dtypes/dtypes.py", line 561, in update_dtype
    new_ordered = dtype._ordered
AttributeError: 'Categorical' object has no attribute '_ordered'
```

cc @TomAugspurger for intent. If a bug maybe we drop this annotation for now and clean up with bug resolution

Reply:
That's a bug in `update_dtype` due to the changes we recently made with ordered cc @jschendel 

The code in there is assuming we have a CategoricalDtype (which has a `_ordered` now), but it can apparently also be a Categorical (or should that raise?), because `is_dtype` allows that.

Reply:
Cool thanks - for now I think OK to remove annotation from this PR then and can be addressed with whatever bug fix is there

Reply:
Opened https://github.com/pandas-dev/pandas/issues/27338 to address this